### PR TITLE
FC-1321 | Add skeleton loading component

### PR DESCRIPTION
**Pull request by zoheiry**
## Purpose
- To start using skeleton loading more consistently.

### Visualization

Screen Shot 2019-04-02 at 1 42 00 PM

### Checklist

- [x] Update changelog

### CODE REVIEW

File: src/animations/skeleton.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { keyframes, css } from 'styled-components';
+
+const commonCss = (color) => css`
+  position: absolute;
+  height: 100%;
+  top: 0;
+  background: ${color || '#f5f7fa'};
+`;
+
+const skeleton = (color = '#f5f7fa') => keyframes`
```
Comment (kirill-kruchkov) [inline src/animations/skeleton.js]: Shall we use a colour from the  `theme` here?

File: src/components/SkeletonLoading/SkeletonLoading.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+);
+
+SkeletonLoading.propTypes = {
+  /** Used to set the width of the skeleton block; preferably this should be left at 100% and
+  the width set by the parent */
+  width: PropTypes.string,
+  /** Use the height prop to define custom height that is not provided by one of the appearances */
+  height: PropTypes.string,
+  animating: PropTypes.bool,
+  appearance: PropTypes.oneOf(['text', 'square', 'header', 'circle']),
```
Comment (kirill-kruchkov) [inline src/components/SkeletonLoading/SkeletonLoading.js]: All other components have `appearance` prop connected to status (like danger, success, etc.). I understand that `appearance` makes sense here, but when you look at several components at once, doesn't it look a bit inconsistent to you? Maybe we can call it `shape` or something like that?

File: src/components/SkeletonLoading/SkeletonLoading.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+```jsx
+<SkeletonLoading baseColor="#333" appearance="text" />
+```
+
+Without animation:
+
+```jsx
+<SkeletonLoading animating={false} appearance="header" />
+```
```
Comment (kirill-kruchkov) [inline src/components/SkeletonLoading/SkeletonLoading.md]: Would be really nice to have an example a bit closer to real world, like you have these loaders on Customers page 🙂

File: src/components/SkeletonLoading/SkeletonLoading.js (hunk 2 above)
Comment (zoheiry) [reply src/components/SkeletonLoading/SkeletonLoading.js]: Yea I like `shape` more, I'll rename it 👍

File: src/components/SkeletonLoading/SkeletonLoading.md (hunk 3 above)
Comment (zoheiry) [reply src/components/SkeletonLoading/SkeletonLoading.md]: True, I'll combine multiple types and create real world example

File: src/components/SkeletonLoading/SkeletonLoading.js (hunk 2 above)
Comment (kirill-kruchkov) [reply src/components/SkeletonLoading/SkeletonLoading.js]: Thank you! Just thinking now... maybe we then should also rename `appearance` to `status` some day for the same reason 😄

**Comment by zoheiry** [comment]
@kirill-kruchkov Added a couple of real world examples as you suggested.
Screen Shot 2019-04-02 at 4 32 31 PM

**Comment by kirill-kruchkov** [comment]
@zoheiry thank you Ali, that looks really cool!

**Pull request merged**